ddb-dm-screen.user.js

```diff
@@ -7,246 +7,201 @@
 // @match        https://www.dndbeyond.com/campaigns/*
 // @grant        none
 // ==/UserScript==
+//
 var $ = window.jQuery;
+
+class Character {
+  constructor(json) {
+    this.json = json;
+    this.stats = new Object();
+
+    this.level = 0;
+    for (var i= 0; i < json.classes.length; i++) {
+        this.level += json.classes[i].level;
+    };
+    this.proficiency = Math.ceil(this.level / 4) + 1;
+
+    var stat = ['str', 'dex', 'con', 'int', 'wis', 'cha'];
+    for(var i = 0; i < 6; i++){
+      if (json.overrideStats[i].value == null) {
+        this.stats[stat[i]] = new Stat(json.stats[i].value + json.bonusStats[i].value, this.proficiency);
+      } 
+      else {
+        this.stats[stat[i]] = new Stat(json.overrideStats[i].value, this.proficiency);
+      }
+    }
+
+    // modifiers
+    var mods = json.modifiers; 
+    [mods.race, mods.class, mods.background, mods.item, mods.feat].forEach(function(m){
+      if (m.type == "bonus") {
+        this.stats[m.subType.substr(0,3)].value += m.value;
+      }
+      else if (m.type == "proficiency") {
+        if(m.subType == 'strength-saving-throws') {
+          this.stats.str.saveProficiency = 1;
+        }
+        else {
+          this.stats[m.subType.substr(0,3)].saveProficiency = true;
```

Comment:
There are other modifiers of type "bonus", ex `passive-perception` given by `observant` feat.

Reply:
I started making most of those changes before you added all the skills to the code.
If you know what to change that would help.

Reply:
is this the part that should be added?
```
default:
                                        if (typeof(character.skills[y.subType]) === "undefined" ) {
                                            character.skills[y.subType] = new Skill(y.subType, character.proficiency, "proficiency");
                                        } else {character.skills[y.subType].setProf("proficiency")}; break;
```

Reply:
Also, if the skills are also displayed in the mobile view I think we should scrape them to avoid future bugs.
What do you think?

Reply:
Sure, that would have a benefit of including bardic `jack of all trades` which is not included in my code. I'll try to write something up.

Reply:
As for this part, I think that quick fix would be to test if `subType` ends with `score`, to remove other bonus types.

Reply:
Extracting skills and modifiers:

```
$('.ct-skills__item').each(function(){console.log($(this).children('.ct-skills__col--skill').text());
console.log($(this).children('.ct-skills__col--modifier').text());})
```